drizzlepac/haputils/product.py

```diff
@@ -657,7 +658,41 @@ def wcs_drizzle_product(self, meta_wcs):
             )
         )
 
-        edp_filenames = [element.full_filename for element in self.edp_list]
+        # Determine if there are any single exposure filter images which 
+        # should NOT be used in the computation of the total detection image in
+        # order to minimize cosmic ray contamination
+        #
+        # Determine the unique filters in the total product and create a
+        # dictionary to hold the filter name and the number of exposures per filter
+        set_of_filters = set([element.filters for element in self.edp_list])
+        count_dict = {key: [] for key in set_of_filters}
+
+        # Loop over the exposure objects to collect the exposures for each filter
+        for expo in self.edp_list:
+            for key in count_dict:
+                if key == expo.filters:
+                    count_dict[key].append(expo.full_filename)
+                    break
```

Comment:
```suggestion
        count_dict = {}

        # Loop over the exposure objects to collect the exposures for each filter
        for expo in self.edp_list:
            count_dict.set_default(expo.filters,[]).append(expo.full_filename))
```

I think the current code is much more complicated than it needs to be.  I removed the `set_of_filters` variable because I think it is not used elsewhere.  If I'm wrong about that, it can be computed from `count_dict` after the loop.